kalite/topic_tools/content_models.py

```diff
@@ -416,19 +427,18 @@ def recurse_availability_up_tree(node, available):
                     recurse_availability_up_tree(parent, available)
 
 
-            for path, update in updates_dict.iteritems():
+            for path, update in content_models:
                 if update:
                     # We have duplicates in the topic tree, make sure the stamping happens to all of them.
-                    items = Item.select().where(Item.path == path)
-                    for item in items:
-                        if item.kind != "Topic":
-                            item_data = unparse_model_data(model_to_dict(item, recurse=False))
-                            item_data.update(update)
-                            item_data = parse_model_data(item_data)
-                            for attr, val in item_data.iteritems():
-                                setattr(item, attr, val)
-                            item.save()
-                            recurse_availability_up_tree(item, update.get("available", False))
+                    item = Item.get().where(Item.path == path)
+                    if item.kind != "Topic":
```

Comment:
@rtibbles we removed the `for` loop here thanks to your tip of `path` being unique. It also turns out that we only save about 3 MB by moving this to the DB, so you were right that we don't save much memory here.